Load remote fonts from the URL’s query string

delucis opened the pull request:
The [Samsa web app](https://github.com/Lorp/samsa) has a nice feature where you can preload a font by setting it in the URL’s query string and I thought it would be cool to have the same feature on Bulletproof, so a designer can provide a link directly to a Bulletproof test tab with their font loaded.

This PR allows loading remote fonts via URLs in the query string, e.g. `/ABCs?preload=https://www.example.com/Font.ttf`. It also supports multiple `preload` parameters. Each will be available in the font drop-down menu and the initially selected font can be set using the `f` parameter like for the default fonts currently, e.g. `/glyphs?preload=FONT_A_URL&preload=FONT_B_URL&f=FONT_A_NAME`. What do you think?

Some considerations:

- Remote fonts must be served over HTTPS, otherwise loading will fail.
- There could be some UI to do this too, e.g. a link icon next to the upload button that allowed input for a font URL.
- I haven’t added any documentation for this yet, as I noticed the `f` parameter was also undocumented, but would happily add some.
- Because this uses the same code as the default font loading (basically replacing the default fonts with the fonts set in the query string), it only loads the fonts when the page first loads — you can’t navigate between fonts like this within the app.

hyvyys commented:
That’s cool! We could document this on the front page and add some GUI, no problem. One thing that bugs me though, is that a vast majority of fonts on CDNs are `woff2`, and [opentype.js](https://github.com/opentypejs/opentype.js) does not support them. I tried to replace it with [fontkit](https://github.com/foliojs/fontkit) or [opentype](https://github.com/bramstein/opentype) but got stuck on making either work with webpack. Any ideas?

delucis commented:
@hyvyys I’ve added some documentation for this — hopefully it’s clear enough even for people not familiar with URL query strings.

I probably won’t have time at least in the near future to look at the font parsing infrastructure, but both those libraries seem to also work server-side, so maybe the issue was some Node.js API not bundling properly? For now I’ve added a note in the documentation saying that `woff2` loading isn’t supported.

hyvyys commented:
Thanks!
I added some GUI for this as well: an URL input on the home page and a new “Add remote font” entry at the bottom of the font dropdown. The query is updated automatically after loading a font, so you can simply copy the URL from the address bar after loading a font.

delucis commented:
@hyvyys Oh, cool! Look forward to sharing some links 😄
